### ISSUE TITLE
Error 400: Bad Request

### ISSUE THREAD

**Issue by NaveenAare** [JuanBindez/pytubefix#332]
from pytubefix import YouTube
from pytubefix.cli import on_progress

url = "https://www.youtube.com/watch?v=777hpBZDCkQ"

yt = YouTube(url, use_oauth=True, allow_oauth_cache=True)

print(yt.title)

ys = yt.streams.get_highest_resolution()
ys.download()

Error:

Traceback (most recent call last):
  File "/root/myyuotube/test.py", line 11, in <module>
    print(yt.title)
          ^^^^^^^^
  File "/root/venv/lib/python3.12/site-packages/pytubefix/__main__.py", line 674, in title
    self._author = self.vid_info.get("videoDetails", {}).get(
                   ^^^^^^^^^^^^^
  File "/root/venv/lib/python3.12/site-packages/pytubefix/__main__.py", line 421, in vid_info
    innertube_response = innertube.player(self.video_id)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/venv/lib/python3.12/site-packages/pytubefix/innertube.py", line 785, in player
    return self._call_api(endpoint, query, self.base_data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/venv/lib/python3.12/site-packages/pytubefix/innertube.py", line 692, in _call_api
    response = request._execute_request(
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/venv/lib/python3.12/site-packages/pytubefix/request.py", line 35, in _execute_request
    return urlopen(request, timeout=timeout)  # nosec
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 215, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 521, in open
    response = meth(req, response)
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 630, in http_response
    response = self.parent.error(
               ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 559, in error
    return self._call_chain(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 492, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 639, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: Bad Request

**Comment by sb0y** [comment]
same issue

```
warp_beacon  |   File "/usr/local/lib/python3.10/dist-packages/warp_beacon/scraper/youtube/abstract.py", line 198, in _download_hndlr
warp_beacon  |     ret_val = func(*args, **kwargs)
warp_beacon  |   File "/usr/local/lib/python3.10/dist-packages/warp_beacon/scraper/youtube/music.py", line 21, in _download
warp_beacon  |     stream = yt.streams.get_audio_only()
warp_beacon  |   File "/usr/local/lib/python3.10/dist-packages/pytubefix/__main__.py", line 632, in streams
warp_beacon  |     self.check_availability()
warp_beacon  |   File "/usr/local/lib/python3.10/dist-packages/pytubefix/__main__.py", line 312, in check_availability
warp_beacon  |     status, messages = extract.playability_status(self.vid_info)
warp_beacon  |   File "/usr/local/lib/python3.10/dist-packages/pytubefix/__main__.py", line 421, in vid_info
warp_beacon  |     innertube_response = innertube.player(self.video_id)
warp_beacon  |   File "/usr/local/lib/python3.10/dist-packages/pytubefix/innertube.py", line 785, in player
warp_beacon  |     return self._call_api(endpoint, query, self.base_data)
warp_beacon  |   File "/usr/local/lib/python3.10/dist-packages/pytubefix/innertube.py", line 692, in _call_api
warp_beacon  |     response = request._execute_request(
warp_beacon  |   File "/usr/local/lib/python3.10/dist-packages/pytubefix/request.py", line 35, in _execute_request
warp_beacon  |     return urlopen(request, timeout=timeout)  # nosec
warp_beacon  |   File "/usr/lib/python3.10/urllib/request.py", line 216, in urlopen
warp_beacon  |     return opener.open(url, data, timeout)
warp_beacon  |   File "/usr/lib/python3.10/urllib/request.py", line 525, in open
warp_beacon  |     response = meth(req, response)
warp_beacon  |   File "/usr/lib/python3.10/urllib/request.py", line 634, in http_response
warp_beacon  |     response = self.parent.error(
warp_beacon  |   File "/usr/lib/python3.10/urllib/request.py", line 563, in error
warp_beacon  |     return self._call_chain(*args)
warp_beacon  |   File "/usr/lib/python3.10/urllib/request.py", line 496, in _call_chain
warp_beacon  |     result = func(*args)
warp_beacon  |   File "/usr/lib/python3.10/urllib/request.py", line 643, in http_error_default
warp_beacon  |     raise HTTPError(req.full_url, code, msg, hdrs, fp)
warp_beacon  | urllib.error.HTTPError: HTTP Error 400: Bad Request
warp_beacon  |
warp_beacon  | During handling of the above exception, another exception occurred:
warp_beacon  |
warp_beacon  | Traceback (most recent call last):
warp_beacon  |   File "/usr/local/lib/python3.10/dist-packages/warp_beacon/scraper/__init__.py", line 130, in do_work
warp_beacon  |     items = actor.download(job.url)
warp_beacon  |   File "/usr/local/lib/python3.10/dist-packages/warp_beacon/scraper/youtube/music.py", line 51, in download
warp_beacon  |     return self._download_hndlr(self._download, url)
warp_beacon  |   File "/usr/local/lib/python3.10/dist-packages/warp_beacon/scraper/youtube/abstract.py", line 218, in _download_hndlr
warp_beacon  |     raise TimeOut(extract_exception_message(e))
warp_beacon  | warp_beacon.scraper.exceptions.TimeOut: Bad Request
```

**Comment by felipeucelli** [comment]
Duplicate #327.

**Comment by JuanBindez** [comment]
The version that corrects this has been released, 8.4.1 and taking advantage of the use of oauth, I made a PR to reset the cache, if you can leave your opinions on whether or not I should launch it, if it will help... I tested it and it made it a lot easier , if you want to test this in the "reset_cache" branch, see the PR on how to use it -> #333

**Comment by sb0y** [comment]
Found solution
```python
yt_opts = {"url": url}
yt_opts["client"] = "TV_EMBED"
yt_opts["use_oauth"] = True
yt_opts["allow_oauth_cache"] = True
yt_opts["token_file"] = "your_session_file.json"
return YouTube(**yt_opts)
```

**Comment by NaveenAare** [comment]
It was fixed in `8.4.1`. Thanks
